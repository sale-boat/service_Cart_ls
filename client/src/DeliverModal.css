.deliverTo{
    margin-top : 10px;
    padding-top : 10px;
    border-top: 1px solid #e7e7e7;
    border-bottom : 1px solid #e7e7e7;
}
 
.locator{
    display: inline-block;
}

.zipCodeText{
    display: inline;
    width: 150px;
    margin-left: 10px;
}

.deliverToZipCode{
    display:flex;
    align-items:center;
}

.locatorimage{
    height: 25px;
    width: 22px;
}

.zipCode{
    color : #0066c0;
    height: 15px;
}

.zipCode:hover{
    color :  #B12704;
    cursor : pointer;
}

.delivery{
    display: none;
    position: fixed;
    z-index: 1;
    padding-top: 150px;
    left: 0;
    top: 0;
    width: 100%;
    height: 100%; 
    overflow: auto;
    background-color: rgba(0,0,0,0.7);
}

.deliveryHeader{
    text-align: left;
    background: linear-gradient(to bottom,#f7f7f7,#eaeaea);
    padding:0px 13px;
    font-weight: bold;
    display : block;
    color: #111;
    font-size: 13px;
    margin: auto;
    width: 25%;
    border: 1px solid #e1e1e1;
    border-top-color: #949494;
    z-index: 100;
    border-radius: 4px 4px 0 0;
}

.myName, .chooseLocation{
   font-weight: bold;
}

.myName{
    margin-bottom: 10px;
}



.deliveryContent{
    background-color: #fff;
    padding:10px 14px;
    display: block;
    width: 25%;
    margin: auto;
}

.deliveryContentOrangebox{
    border: 1px solid #fab877;
    padding: 10px;
    background-color: #fef8f2;
    margin: 10px 0px;
}

.address{
    margin-top: 10px;
    color: #898887;
    font-size: 13px;
    font-weight: bold;
}

.addressBook{
    color : #0066c0;
}

.addressBook:hover{
   text-decoration: underline;
   color :  #B12704;
   cursor : pointer;
}

.UsZip{
    text-align: center;
    position: relative;
    z-index: 100;
    background-color: white;
    font-size: 13px;
    color: #949494;
}


.UsZip:before{
    content: '';
    position: absolute;
    top:50%;
    left: 0;
    overflow: hidden;
    width: 30%;
    border-top: 1px solid #f1f1f1;
    height: 1px;
    z-index: -100;
}

.UsZip:after{
    content: '';
    position: absolute;
    top:50%;
    right: 30;
    overflow: hidden;
    width: 30%;
    border-top: 1px solid #f1f1f1;
    height: 1px;
    z-index: -100;
}

.or{
    text-align: center;
    position: relative;
    z-index: 100;
    background-color: white;
    font-size: 13px;
    color: #949494;
}

.or:before{
    content: '';
    position: absolute;
    top:50%;
    left: 0;
    overflow: hidden;
    width: 45%;
    border-top: 1px solid #f1f1f1;
    height: 1px;
    z-index: -100;
}

.or:after{
    content: '';
    position: absolute;
    top:50%;
    right: 55;
    overflow: hidden;
    width: 45%;
    border-top: 1px solid #f1f1f1;
    height: 1px;
    z-index: -100;
}

.options{
    -webkit-appearance: none;
    background: url(https://s3.us-east-2.amazonaws.com/chrisfakephotos/amazonarrow.png);
    background-repeat: no-repeat;
    background-position: right center;
    width: 100%;
    height: 29px;
    border: #ADB1C9 #A2A6AC #8D9096;
    background-color: #f1f2f5;
    padding-left: 10px;
    font-size: 13px;
    color:#515151;
    font-weight: 400;
}

.options option{
    font-size: 13px;
    width: 100%;
    height: 29px;
}

.deliveryFooter{
    text-align: right;
    background-color: #f8f8f8;
    padding:10px 13px;
    font-weight: bold;
    display : block;
    color: #111;
    font-size: 13px;
    margin: auto;
    width: 25%;
    border: 1px solid #e3e3e3;
    z-index: 100;
    border-radius: 0 0 4px 4px;
}

.done{
   background: linear-gradient(to bottom,#f7dfa5,#f0c14b);
   border-color: #a88734 #9c7e31 #846a29;
   border-radius: 3px;
   padding: 8px 10px;
   font-size: 13px;
}

.textInput{
    border-radius: 3px;
    width: 65%;
    height: 30px;
}

.textSubmit{
    border-radius: 3px;
    height: 38px;
    width: 90px;
    margin-left: 5px;
    font-size: 13px;
    background: linear-gradient(to bottom,#f7f8fa,#e7e9ec);
    border-color: #ADB1B8 #A2A6AC #8D9096;
}