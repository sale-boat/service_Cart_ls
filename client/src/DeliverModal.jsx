import React from 'react';
import deliverModal from './DeliverModal.css';

class DeliverModal extends React.Component {
  constructor(props) {
    super(props);
    this.showModal = this.showModal.bind(this);
    this.hideModal = this.hideModal.bind(this);
    this.state = {
      modalShow: false,
    };
  }

  showModal() {
    this.setState({ modalShow: true });
  }

  hideModal(event) {
    if (event.target.className === deliverModal.delivery) {
      this.setState({ modalShow: false });
    }
  }

  render() {
    const { modalShow } = this.state;
    const style = { display: modalShow ? 'block' : 'none' };

    return (
      <div className={deliverModal.deliverTo}>
        <div className={deliverModal.delivery} style={style} onClick={this.hideModal}>
          <div className={deliverModal.deliveryHeader}>
            <span className={deliverModal.chooseLocation}><h3>Choose your location</h3></span>
          </div>
          <div className={deliverModal.deliveryContent}>
            <span>Delivery options and delivery speeds may vary for different locations</span>
            <div className={deliverModal.deliveryContentOrangebox}>
              <span>
                <span className={deliverModal.myName}>
                 christopher tso- chicago
                 {' '}
                - 60632
                </span>  
                <br />
                <div className={deliverModal.address}>
                  <span >Default address</span>
                </div>
              </span>
            </div>
            <span className={deliverModal.addressBook}>Manage address book</span>
            <br />
            <p className={deliverModal.UsZip}>or enter a US zip code</p>
            <input type="text" className={deliverModal.textInput} /> 
            {' '}
            <input type="submit" value="Apply" className={deliverModal.textSubmit} />
            <p className={deliverModal.or}>or</p>
            <select className={deliverModal.options}>
              <option>Ship outside the US</option>
            </select>
          </div>
          <div className={deliverModal.deliveryFooter}>
            <button type="button" className={deliverModal.done}>Done</button>
          </div>
        </div>
        <div className={deliverModal.deliverToZipCode}>
          <div className={deliverModal.locator}>
            <img className={deliverModal.locatorimage} src="https://s3.us-east-2.amazonaws.com/chrisfakephotos/Location.png" alt="locator" />
          </div>
          <div className={deliverModal.zipCodeText}>
            <span className={deliverModal.zipCode} onClick={this.showModal}>
              {' '}
              {'Deliver to Chicago 60632'}
            </span>
          </div>
        </div>
      </div>
    );
  }
}

export default DeliverModal;
