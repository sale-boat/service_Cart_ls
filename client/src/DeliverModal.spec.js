import React from 'react';
import { mount, shallow } from 'enzyme';
import DeliverModal from './DeliverModal.jsx';

describe('Deliver Component', () => {
  it('should render', () => {
    const deliver = shallow(<DeliverModal />);
    expect(deliver).toMatchSnapshot();
  });

  it('should call show modal when clicked and modal should exist', () => {
    const showModal = jest.spyOn(DeliverModal.prototype, 'showModal');
    const wrapper = mount(<DeliverModal /> );
    const zipCode = wrapper.find('.zipCode');
    zipCode.simulate('click');
    expect(showModal).toHaveBeenCalledTimes(1);
    expect(wrapper.find('.delivery').exists()).toEqual(true);
  });

  it('should set the showModal state to true when Deliver To is clicked', () => {
    const wrapper = shallow(<DeliverModal />);
    const zipCode = wrapper.find('.zipCode');
    zipCode.simulate('click');
    expect(wrapper.state().modalShow).toEqual(true);
  })
});