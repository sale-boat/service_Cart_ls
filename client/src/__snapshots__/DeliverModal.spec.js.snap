// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deliver Component should render 1`] = `
<div
  className="deliverTo"
>
  <div
    className="delivery"
    onClick={[Function]}
    style={
      Object {
        "display": "none",
      }
    }
  >
    <div
      className="delivery-header"
    >
      <span
        className="chooseLocation"
      >
        <h3>
          Choose your location
        </h3>
      </span>
    </div>
    <div
      className="delivery-content"
    >
      <span>
        Delivery options and delivery speeds may vary for different locations
      </span>
      <div
        className="delivery-content-orangebox"
      >
        <span>
          <span
            className="myName"
          >
            christopher tso- chicago
             
            - 60632
          </span>
          <br />
          <div
            className="address"
          >
            <span>
              Default address
            </span>
          </div>
        </span>
      </div>
      <span
        className="addressBook"
      >
        Manage address book
      </span>
      <br />
      <p
        className="UsZip"
      >
        or enter a US zip code
      </p>
      <input
        className="textInput"
        type="text"
      />
       
      <input
        className="textSubmit"
        type="submit"
        value="Apply"
      />
      <p
        className="or"
      >
        or
      </p>
      <select
        className="options"
      >
        <option>
          Ship outside the US
        </option>
      </select>
    </div>
    <div
      className="delivery-footer"
    >
      <button
        className="done"
        type="button"
      >
        Done
      </button>
    </div>
  </div>
  <div
    className="deliverToZipCode"
  >
    <div
      className="locator"
    >
      <img
        alt="locator"
        className="locatorimage"
        src="https://s3.us-east-2.amazonaws.com/chrisfakephotos/Location.png"
      />
    </div>
    <div
      className="zipCodeText"
    >
      <span
        className="zipCode"
        onClick={[Function]}
      >
         
        Deliver to Chicago 60632
      </span>
    </div>
  </div>
</div>
`;
